### ISSUE TITLE
Feature Request: Generic rule looking for pattern/regex

### ISSUE THREAD

**Issue by fe60** [adrienverge/yamllint#392]
I want to avoid or enforce the usage of certain strings in texts/descriptions.
A generic `pattern` rule would allow me to do so. It should report either the existence or the absence of strings matching a pattern/regex.
```yaml
rules:
  pattern:
    - name: forbidden
      message: don't use "WTF" or "wtf"
      pattern: '([wW][tT][fF])*'
      apply_to: [key_1, key_2]
    - name: polite
      message: you missed something
      pattern: (P|p)lease
      report_absence: True # default: False
      apply_to: key_x
```
Is this a valid use case/approach?
Does the general structure of yamllint allow the implementation of such a rule [1]?

[1] I'm new to Python and new to yamllint. Still I'm willing to contribute such a rule (or at least give it a try), as it will be a good learning opportunity. But a rating about the feasibility is welcome.

**Comment by adrienverge** [comment]
Hello, thanks for raising this topic. Yamllint is a linter: it tries to validate the form, not the content. Checking *string values*, for example requiring strings to match / not match certain patterns is beyond what yamllint is intended to do. I suggest writing a custom tool for this purpose.

**Comment by fe60** [comment]
Valid point. I will go for a separate tool/script.
